src/config/libinput.c

```diff
@@ -26,7 +26,7 @@ libinput_category_init(struct libinput_category *l)
 enum device_type
 get_device_type(const char *s)
 {
-	if (!s) {
+	if (!s || !*s) {
 		return DEFAULT_DEVICE;
 	}
```

Comment:
This is technically not needed, none of the following would match and we'd return `DEFAULT_DEVICE` anyway.